### DISCUSSION TITLE
关于 0-RTT 建立连接

### DISCUSSION THREAD

**Discussion by nixwl** [Tencent/tquic#152] [Q&A (问题和回答)] [answered]
我希望使用 示例程序 tquic_server、 tquic_client 验证 0-RTT 建立连接：（在tquic_client 的 early_data、 session-file 均以设置）
- 首先，我运行了tquic_server（用 wireshark 抓取 quic 包），
- 然后，运行一次 tquic_client ，此时 wireshark 抓取到 1-RTT 包，
- 接着，我试图再运行一次 tquic_client ，使其完成0-RTT建立连接，

但是在 qlog 里没有检查到 0-RTT 包的传输， wirshark 抓出的包看起来也不太像 0-RTT 建立连接。

请问是我的测试思路存在问题吗？麻烦了

**Comment by iyangsj** [comment, marked as answer]
测试思路没有问题。

TQUIC库目前[支持0-RTT](https://tquic.net/zh/docs/tutorial/rust/#0rtt%E5%BB%BA%E7%AB%8B%E8%BF%9E%E6%8E%A5)。但tquic_client示例程序比较简单，并未使用TQUIC库所有支持功能，tquic_client示例暂未实现0RTT
注：tquic_client 目前仅[在握手完成时](https://github.com/Tencent/tquic/blob/develop/tools/src/bin/tquic_client.rs#L1274-L1275)，而非连接创建时尝试发送请求。

**Reply by nixwl** [reply]
好的，非常感谢

**Comment by iyangsj** [comment]
#153
